File: app/src/main/java/io/github/tobyhs/weatherweight/forecast/ForecastModule.java
package io.github.tobyhs.weatherweight.forecast;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Dagger module for {@link ForecastActivity}
 */
@Module
public abstract class ForecastModule {
    @ContributesAndroidInjector
    abstract ForecastActivity contributeForecastActivityInjector();
}

File: app/src/main/java/io/github/tobyhs/weatherweight/forecast/ForecastModule.kt
package io.github.tobyhs.weatherweight.forecast

import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityRetainedComponent

/**
 * Dagger module for [ForecastActivity]
 */
@Module
@InstallIn(ActivityRetainedComponent::class)
abstract class ForecastModule
